utils/csv2img-exif

```diff
@@ -0,0 +1,179 @@
+
+def fix_webp(target):
+    return str(check_output(["cwebp", "-quiet", "mt", "-z", "9", "-metadata", "exif", target, "-o", target]), 'UTF-8')
+
+
+##
+# program logic start
+##
+
+args = argparse.ArgumentParser()
+args.add_argument("--repo", default="..", help="The directory of the Wesnoth repository to run this script against.")
+args.add_argument("--input", default="copyrights.csv", help="The file to read the existing copyright data from.")
+args.add_argument("--output", default="csv2img-exif_output.csv", help="The file to write the results of this script to.")
+options = args.parse_args()
+os.chdir(options.repo)
+
+
+csv_data = {}
+# Too few fields
+missing_fields = []
+# Too many fields, possibly due to an unquoted comma
+extra_fields = []
+# previously untracked images
+added = []
+# lacking something in either the license or author fields
+incomplete = []
+unchanged = []
+removed = []
+unknown_licenses = []
+
+
+with open(options.input, encoding="utf-8") as csvfile:
+    reader = csv.reader(csvfile)
+    previous_file = ""
+    for row in reader:
+        if row[0] == "Date":
+            continue
+
+        file = row[1]
+        previous_file = file
+
+        if not os.path.exists(file):
+            removed.append(file)
+            continue
+
+        csv_data[file] = row
+
+with contextlib.suppress(FileNotFoundError):
+    os.remove(options.output)
+
+# only transfer the info from complete, properly formatted CSV rows (unchanged)
+# but keep track of all the rest for summary output
+for root, _, files in os.walk(options.repo):
+    for filename in files:
+        filetype = Path(filename).suffix
+        if filetype == ".png" or filetype == ".jpg":
+            file = os.path.normpath(os.path.join(root, filename))
+            file = re.sub(r'../wesnoth/', '', file)
+
+            if not file in csv_data:
+                added.append(["NO_DATA", file, "NO_DATA", "NO_DATA", "NEW", "NO_DATA", "NO_DATA"])
+            elif len(csv_data[file]) < 7:
+                missing_fields.append(csv_data[file])
+            elif len(csv_data[file]) > 7:
+                extra_fields.append(csv_data[file])
+            elif csv_data[file][2].strip() == "" or csv_data[file][3].strip() == "":
+                incomplete.append(csv_data[file])
+            elif not csv_data[file][2] in known_licenses:
+                unknown_licenses.append(csv_data[file][2])
+                incomplete.append(csv_data[file])
+            else:
+                unchanged.append(csv_data[file])
+                write_exif(file,csv_data[file][0],csv_data[file][2],csv_data[file][3],csv_data[file][4])
+
+        elif filetype == ".webp":
+            file = os.path.normpath(os.path.join(root, filename))
+            file = re.sub(r'../wesnoth/', '', file)
+
+            if not file in csv_data:
+                added.append(["NO_DATA", file, "NO_DATA", "NO_DATA", "NEW", "NO_DATA", "NO_DATA"])
```

Comment:
This should at a minimum leave fields 3 and 4 empty or the check for empty further down will not consider these entries as incomplete.